atsign-foundation/at_libraries
# fix: stricter regex for reserved keys

Closes https://github.com/atsign-foundation/at_libraries/issues/437

**- What I did**
- modified the regex for reserved keys such that there is an explicit check for keys (including public/private tag and pre/postfix atsigns).
- added a new variable in AtConstants for the key that stores the atsign's blocklist.
- previously public hidden keys (public:__somekey@atsign) was considered reserved key, which it's not. With the current changes, it will NOT be considered a reserved key.

**- How I did it**
- Divided the keys into 2 groups:
1. Keys that do not contain atsign (e.g. privatekey:at_pkam_publickey, _latestNotificationId)
2. Keys that contain atsign as prefix/postfix (e.g. '@'bob:shared_key'@'alice, public:publickey'@'alice)
- These twogroups have their own regex and all these regexes are joined using an "OR" operator
- Made sure that the visibility of each key (public/private/shared) is also explicitly mentioned in the regex

- Introduced a new method 'isPartalMatch()' in class Regexutil that does not require a full string match (only for reserved keys). This is required as the public/private part of the key is a lookahead and will not be matched as part of the key

**- How to verify it**
- positive and negative tests have been added to ensure that test reserved keys explicitly to ensure proper classification

**- Description for the changelog**

fix: stricter regex for reserved keys
